src/server/dfly_main.cc

```diff
@@ -462,15 +462,24 @@ bool UpdateResourceLimitsIfInsideContainer(io::MemInfoData* mdata, size_t* max_t
   /* Update memory limits */
 
   // Start by reading global memory limits
-  constexpr auto base_mem = "/sys/fs/cgroup/memory"sv;
-  read_mem(StrCat(base_mem, "/memory.limit_in_bytes"), &mdata->mem_total);
-  read_mem(StrCat(base_mem, "/memory.max"), &mdata->mem_total);
+  auto parse_limits = [&](std::string_view base_mem) {
+    read_mem(StrCat(base_mem, "/memory.limit_in_bytes"), &mdata->mem_total);
+    read_mem(StrCat(base_mem, "/memory.max"), &mdata->mem_total);
+  };
+
+  // For v1
+  constexpr auto base_mem_v1 = "/sys/fs/cgroup/memory"sv;
```

Comment:
This is wrong. There are no `memory.max` in `root cgroup`. That means that always, even before this PR the `read_mem(StrCat(base_mem, "/memory.limit_in_bytes"), &mdata->mem_total);` would never read anything